### PR TITLE
feat: add dbaas logs opensearch alias and index resources

### PR DESCRIPTION
**Pull request by marie-j**
# Description

- Add Dbaas logs Output opensearch alias resource
- Add Dbaas logs Output opensearch index resource

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: `make testacc TESTARGS="-parallel=1 -run DbaasLogs"`

**Test Configuration**:
* Terraform version: `terraform version`: Terraform vx.y.z
* Existing HCL configuration you used: 
```hcl
resource "ovh_dbaas_logs_output_opensearch_alias" "alias" {
 service_name = "ldp-xxx"
 description = "aa"
 suffix        = "aa"
}

resource "ovh_dbaas_logs_output_opensearch_index" "index" {
  service_name = "ldp-xxx"
  description = "aa"
  suffix        = "aa"
  nb_shard = 1
}
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or issues
- [x] I have added acceptance tests that prove my fix is effective or that my feature works
- [x] New and existing acceptance tests pass locally with my changes
- [ ] I ran succesfully `go mod vendor` if I added or modify `go.mod` file

### CODE REVIEW

File: ovh/data_dbaas_logs_output_opensearch_index.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		"/dbaas/logs/%s/output/opensearch/index",
+		url.PathEscape(serviceName),
+	)
+	if err := config.OVHClient.Get(endpoint, &res); err != nil {
+		return diag.Errorf("Error calling Get %s:\n\t %q", endpoint, err)
+	}
+	indexes := []*DbaasLogsOutputOpensearchIndex{}
+
+	for _, id := range res {
+		log.Printf("[DEBUG] Will read dbaas logs output opensearch index id : %s/%s", serviceName, id)
```
Comment (deathiop) [inline ovh/data_dbaas_logs_output_opensearch_index.go:98]:
```suggestion
		log.Printf("[DEBUG] Will read dbaas logs output opensearch index id: %s/%s", serviceName, id)
```

File: ovh/data_dbaas_logs_output_opensearch_index.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			url.PathEscape(serviceName),
+			url.PathEscape(id),
+		)
+
+		index := &DbaasLogsOutputOpensearchIndex{}
+		if err := config.OVHClient.Get(endpoint, &index); err != nil {
+			return diag.Errorf("Error calling Get %s:\n\t %q", endpoint, err)
+		}
+
+		log.Printf("[INFO]Comparing : %s ? %s",
```
Comment (deathiop) [inline ovh/data_dbaas_logs_output_opensearch_index.go:111]:
```suggestion
		log.Printf("[INFO] Comparing: %s ? %s",
```

Shouldn't it be a debug?

File: ovh/resource_dbaas_logs_output_opensearch_alias.go [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceDbaasLogsOutputOpensearchAliasImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenID := d.Id()
+	splitID := strings.SplitN(givenID, "/", 2)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias.go:111]:
```suggestion
	serviceName, id, ok := strings.Cut(givenID, "/")
```

File: ovh/resource_dbaas_logs_output_opensearch_alias.go [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+		return nil, fmt.Errorf("Import Id is not service_name/id formatted")
+	}
+	serviceName := splitID[0]
+	id := splitID[1]
+	d.SetId(id)
+	d.Set("service_name", serviceName)
+
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias.go:122]: Nitpick, easier to `return []*schema.ResourceData{d}, nil`

File: ovh/resource_dbaas_logs_output_opensearch_alias.go [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+	}
+
+	id := op.AliasId
+	if id == nil {
+		return diag.Errorf("Alias Id is nil. This should not happen: operation is %s/%s", serviceName, res.OperationId)
+	}
+
+	d.SetId(*id)
+
+	indexes := d.Get("indexes").([]interface{})
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias.go:155]:
```suggestion
	indexes := d.Get("indexes").([]string)
```

and envoid the next `index.(string)`?

File: ovh/resource_dbaas_logs_output_opensearch_alias_test.go [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+		streams := []string{}
+		if err := config.OVHClient.Get(endpoint, &streams); err != nil {
+			return fmt.Errorf("Error calling Get %s:\n\t %q", endpoint, err)
+		}
+
+		for _, s := range streams {
+			res := &DbaasLogsOperation{}
+			ctx := context.Background()
+			err = retry.RetryContext(ctx, 5*time.Minute, func() *retry.RetryError {
+				log.Printf("[INFO] Will delete detach stream from opensearch alias : %s/%s", serviceName, id)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias_test.go:67]:
```suggestion
				log.Printf("[INFO] Will delete detach stream from opensearch alias: %s/%s", serviceName, id)
```

File: ovh/resource_dbaas_logs_output_opensearch_alias_test.go [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+					return retry.RetryableError(err)
+				}
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		log.Printf("[DEBUG] Will read dbaas logs output opensearch alias id : %s/%s", serviceName, id)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias_test.go:82]:
```suggestion
		log.Printf("[DEBUG] Will read dbaas logs output opensearch alias id: %s/%s", serviceName, id)
```

File: ovh/resource_dbaas_logs_output_opensearch_alias_test.go [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+		}
+
+		if !strings.HasPrefix(alias.Description, test_prefix) {
+			continue
+		}
+
+		res := &DbaasLogsOperation{}
+		ctx := context.Background()
+		err = retry.RetryContext(ctx, 5*time.Minute, func() *retry.RetryError {
+			log.Printf("[INFO] Will delete dbaas logs output graylog stream : %s/%s", serviceName, id)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_alias_test.go:102]:
```suggestion
			log.Printf("[INFO] Will delete dbaas logs output graylog stream: %s/%s", serviceName, id)
```

File: ovh/resource_dbaas_logs_output_opensearch_index.go [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+				Computed:    true,
+				Description: "Operation last update",
+			},
+		},
+	}
+}
+
+func resourceDbaasLogsOutputOpensearchIndexImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenID := d.Id()
+	splitID := strings.SplitN(givenID, "/", 2)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_index.go:94]: `strings.Cut`

File: ovh/resource_dbaas_logs_output_opensearch_index_test.go [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		if !strings.HasPrefix(Index.Description, test_prefix) {
+			continue
+		}
+
+		res := &DbaasLogsOperation{}
+		ctx := context.Background()
+		err = retry.RetryContext(ctx, 5*time.Minute, func() *retry.RetryError {
+			log.Printf("[INFO] Will delete dbaas logs output graylog stream : %s/%s", serviceName, id)
```
Comment (deathiop) [inline ovh/resource_dbaas_logs_output_opensearch_index_test.go:72]:
```suggestion
			log.Printf("[INFO] Will delete dbaas logs output graylog stream: %s/%s", serviceName, id)
```

File: ovh/resource_dbaas_logs_output_opensearch_alias.go (hunk 5 above)
Comment (marie-j) [reply ovh/resource_dbaas_logs_output_opensearch_alias.go:155]:
```
panic: interface conversion: interface {} is []interface {}, not []string
```
I suppose this is due to this declaration https://github.com/ovh/terraform-provider-ovh/commit/e7dbabe0cce487816643676baf4b4e8d080c1b0c#diff-c62ab4e0d2d514494a5e0c1f4e955ccdf85a59d2fcd1b2b59ca14f7e32b8e79dR88

File: website/docs/r/dbaas_logs_output_opensearch_index.html.markdown [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+* `created_at` - Index creation
+* `current_size` - Current index size (in bytes)
+* `description` - Index description
+* `index_id` - Index ID
+* `is_editable` - Indicates if you are allowed to edit entry
+* `max_size` - Maximum index size (in bytes)
+* `name` - Index name
+* `nb_shard` - Number of shard
+* `updated_at` - Index last update
+
```
Comment (amstuta) [inline website/docs/r/dbaas_logs_output_opensearch_index.html.markdown:41]:
```suggestion
```

**Review by amstuta — APPROVED** [review]

**Pull request merged**